### PR TITLE
Add support for LIKE and STARTS_WITH operators

### PR DESCRIPTION
**Pull request by edhollinghurst**
This PR adds support for use of the LIKE and STARTS_WITH operators when adding a field condition to a query.

### CODE REVIEW

**Comment by amitaibu** [comment]
Thanks. This will also require a (simple) simpleTest :)

**Comment by e0ipso** [comment]
We also need support for those operators on entity properties, not just fields.

**Comment by edhollinghurst** [comment]
When I looked at this again I realised that the LIKE and STARTS_WITH operators are already handled correctly in RestfulDataProviderEFQ.php for both fields and properties. It's just a case of enabling these operators in RestfulBase.php.

I've added a couple of tests for these operators to RestfulListTestCase.test.

**Comment by e0ipso** [comment]
@edhollinghurst actually by enabling it in `RestfulBase` you are enabling it as well for all of the data providers. Probably this will fail for other data providers, like for instance `RestfulDataProviderDbQuery`.

**Comment by edhollinghurst** [comment]
I've now taken out the change from `RestfulBase` and overridden the function in `RestfulDataProviderEFQ`.

I also made a small correction to the error message that's thrown if the operator is not allowed. `$operators` was being passed as the token when this should be `$operator`.

**Comment by edhollinghurst** [comment]
@mateu-aguilo-bosch Is there anything more you need me to do in terms of this?

File: plugins/restful/RestfulDataProviderEFQ.php [hunk 1]
```diff
@@ -209,6 +209,34 @@ protected function getColumnFromProperty($property_name) {
...
+  protected static function isValidOperatorsForFilter(array $operators) {
+    $allowed_operators = array(
+      '=',
+      '>',
+      '<',
+      '>=',
+      '<=',
+      '<>',
+      '!=',
+      'IN',
```
Comment (amitaibu) [inline plugins/restful/RestfulDataProviderEFQ.php:223]: There's also

```
+      'CONTAINS',
+      'IN',
+      'NOT IN',
+      'STARTS_WITH',
```

btw, the file chmod should be changed back to 644

**Comment by amitaibu** [comment]
Should we also add this to `RestfulBase.php`?

This needs a re-roll.

**Comment by e0ipso** [comment]
> Should we also add this to RestfulBase.php?

I don't think that we should. Those apply to all of the data providers, but only some of them support that list of operators.

**Comment by e0ipso** [comment]
I'm going to close this issue to unclutter the PR view. Feel free to re-open if you pick up this work again.

Thanks for everything!

**Pull request closed without merging**